js/server/modules/@arangodb/foxx/router/swagger-context.js

```diff
@@ -655,6 +655,9 @@ function swaggerifyBody (schema, optional) {
 
 function joi2schema (schema) {
   if (!schema.isJoi) {
+    if (typeof schema !== "object") {
+      return;
+    }
     return schema;
   }
   return joiToJsonSchema(schema);
```

Comment:
```
  if (typeof schema !== "object") {
      return;
  else if (!schema.isJoi) { 
      return schema;
  } else {
      return joiToJsonSchema(schema);
```

Reply:
Isn't a variable typo should precede a member call?

Reply:
I initially checked for `boolean` but extended it later. It doesn't matter if `schema.isJoi` is accessed first, JavaScript simply returns a value of `undefined` in this case.

I don't think the behavior would be any different with your version but it's JavaScript... On the other hand, the existing code doesn't cover certain corner cases anyway. For example, a more water-tight check would not use `typeof` but test for null identity, handle the possibility of an Object with a null prototype, and compare the object's prototype to that of the base Object prototype to cover e.g. `new Boolean()` which has the type `object` and not `boolean`.

Reply:
@Simran-B you don't need to do any of this manually. We have `lodash` as a dep, I think this might be the right place to use it

Reply:
even without lodash, I'd prefer moving the object check out, & doing it before `schema.isJoi` check. Otherwise this same question will be asked in the future

Reply:
I mean sure. But note that there's more code that only checks for the typeof string, missing the same corner cases, and it's not worth fixing as this is Foxx.